### PR TITLE
Bump rspec-rails to 3.4.1

### PR DESCRIPTION
**Pull request by boffbowsh**
This is required for compatibility with Rails 4.1.14.1
Some rspec bumps are also needed to support this.

This is a replacement of #34, rspec-rails 3.4.1 wasn't released when that PR was raised.

/cc @jennyd @mattbostock

### CODE REVIEW

**Comment by boffbowsh** [comment]
(Tests are passing locally)

File: Gemfile.lock [hunk 1]
```diff
@@ -475,3 +476,6 @@ DEPENDENCIES
   webmock
   xmpp4r
   yajl-ruby
+
+BUNDLED WITH
+   1.10.6
```
Comment (mattbostock) [inline Gemfile.lock]: Should this be included?

**Comment by mattbostock** [comment]
One comment about the `BUNDLED WITH` comment, but otherwise merge when ready.

File: Gemfile.lock (hunk 1 above)
Comment (benlovell) [inline Gemfile.lock]: Arbitrary bundler version ahoy! :anchor:  https://github.com/alphagov/govuk-puppet/blob/e82895dbdc1d8432c9cbcfe252a420c1bb591922/modules/govuk_rbenv/manifests/all.pp#L20

**Pull request merged**